linux/memory_hot_add_basic/memory_hot_add_basic.yml

```diff
@@ -14,38 +14,64 @@
   tasks:
     - name: "Test case block"
       block:
+        # Except for VM guest_id like 'freebsd.*Guest' not supporting memory hot-add,
+        # the other guest OS compatible with ansible-vsphere-gos-validation support memory hot-add
+        - name: "Skip test case '{{ ansible_play_name }}' due to VM not support memory hot-add"
+          include_tasks: ../../common/skip_test_case.yml
+          vars:
+            skip_msg: "Skip test case '{{ ansible_play_name }}' because memory hot-add is not supported by guest id {{ vm_guest_id }}"
+            skip_reason: "Not Supported"
+          when:
+            - vm_guest_id is defined
+            - vm_guest_id is match('freebsd.*')
+
         - name: "Test setup"
           include_tasks: ../setup/test_setup.yml
+          vars:
+            create_test_log_folder: true
 
-        - name: Set fact of VM initial memory size to 2048MB before hotadd test
+        - name: "Set fact of VM initial memory size to 2048MB before hotadd test"
           ansible.builtin.set_fact:
             vm_initial_mem_mb: 2048
-        - include_tasks: generate_mem_hot_add_list.yml
 
-        - include_tasks: ../../common/skip_test_case.yml
+        - name: "Generate memory size list for hot-add"
+          include_tasks: generate_mem_hot_add_list.yml
+
+        - name: "Test case is blocked due to empty memory size list"
+          include_tasks: ../../common/skip_test_case.yml
           vars:
-            skip_msg: "Test case '{{ ansible_play_name }}' is blocked because memory hotadd test value list is empty"
+            skip_msg: "Test case '{{ ansible_play_name }}' is blocked because memory hot-add test value list is empty"
             skip_reason: "Blocked"
           when: memory_hotadd_size_list | length == 0
 
-        - include_tasks: ../../common/skip_test_case.yml
+        # From guest OS, Debian for 32bit and FreeBSD for 32bit and 64bit don't support memory hot-add.
```

Comment:
There is `vm_guest_id is match('freebsd.*')` above, can this part also be moved as above to match guest ID?

Reply:
`guest_os_ansible_distribution` is set in `test_setup.yml` for existing VM, so it can't be moved above. I will file another PR to move out `get_linux_system_info.yml` from `test_setup.yml`, and then here can be improved.